src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs

```diff
@@ -135,6 +135,9 @@ internal void ReleaseConnection()
                 {
                     if (!_isClosed && _tcpClient != null)
                     {
+                        // Gracefully close the transmission channel
```

Comment:
If you make the Socket non-blocking (`Socket.Blocking`), the TCP send buffer can't cause the QuitCommand.Send to block.

Reply:
We should wait the reply for spec